src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -39,11 +40,40 @@ public List<Station> orderStationsOfLine() {
         return stations;
     }
 
+    public void removeStationInSection(Station station) {
+        validateDeleteStationInSection(station);
+
+        Optional<Section> optionalUpStation = findByUpStation(station);
+        Optional<Section> optionalDownStation = findByDownStation(station);
+
+        optionalUpStation.ifPresent(section -> this.sections.remove(section));
+        optionalDownStation.ifPresent(section -> this.sections.remove(section));
+
+        if (optionalDownStation.isPresent() && optionalUpStation.isPresent()) {
+            Section upSection = optionalDownStation.get();
+            Section downSection = optionalUpStation.get();
+
+            Distance distance = Distance.sumOf(upSection.getDistance(), downSection.getDistance());
+            this.add(new Section(upSection.getUpStation(), downSection.getDownStation(), distance));
+        }
+    }
+
     public int size() {
         return this.sections.size();
     }
 
-    private void validateSection(Section section) {
+    private void validateDeleteStationInSection(Station station) {
+        if (this.canNotDelete()) {
+            throw new IllegalArgumentException("노선에 남은 구간이 하나이기 떄문에 삭제할 수 없습니다.");
+        }
+
+        List<Section> sections = findAllByStation(station);
+        if (sections.size() == 0) {
```

Comment:
```suggestion
        if (sections.size().isEmpty()) {
```
isEmpty()나 CollectionUtils.isEmpty(List)로 바꾸시는걸 추천드립니다.
전자는 가독성 면에서 훨씬 낫고 후자는 NPE 발생 가능성까지 차단할 수 있습니다.